server/api/externalapi.ts

```diff
@@ -32,13 +32,27 @@ class ExternalAPI {
       this.fetch = fetch;
     }
 
-    this.baseUrl = baseUrl;
-    this.params = params;
+    const url = new URL(baseUrl);
+
     this.defaultHeaders = {
       'Content-Type': 'application/json',
       Accept: 'application/json',
+      ...((url.username || url.password) && {
+        Authorization: `Basic ${Buffer.from(
+          `${url.username}:${url.password}`
+        ).toString('base64')}`,
+      }),
```

Comment:
To be more clear, WDYT of something like this instead?
```suggestion
          Authorization:
            url.username || url.password
              ? `Basic ${Buffer.from(
                  `${url.username}:${url.password}`
                ).toString('base64')}`
              : undefined,
```

Reply:
Yeah I guess it is more concise and clear 🤔
Should I change it to this?

Reply:
IMO that's better, but it's a small thing